### PR TITLE
[421] Environment-aware Explorer URLs

### LINKED ISSUE #421: Cow.swap points to the the wrong explorer URL
**Describe the bug**
All environments of cow swap are pointing to DEV environment. 

**How to Reproduce**
1. Go to http://cow.trade/
2. Do a trade
3. See the link to the explorer, it's pointing to https://protocol-explorer.dev.gnosisdev.com

**Expected behavior**
Each environment points to a different explorer:

* PRODUCTION: cow.trade, should point to https://gnosis-protocol.io 
* STAGING: https://cow-trade.staging.gnosisdev.com, should point to https://protocol-explorer.staging.gnosisdev.com/
* DEV: https://cow-trade.dev.gnosisdev.com, should point to https://protocol-explorer.dev.gnosisdev.com/

### PR DESCRIPTION
**Pull request by W3stside**
Closes #421 

Checks the environment and uses `.env.production` urls + `chainId` to construct `explorer` URLs correctly.

## QA Testing
:warning: _right now this can only best tested easily once merged in `develop` or `master` as the available URLs here won't trigger anything_

1. make orders and check the toast notification URLs are correct depending on what URL and chainId you are on
  ```bash
cow-trade.dev = https://protocol-explorer.dev.gnosisdev.com
cow-trade.staging = https://protocol-explorer.staging.gnosisdev.com
cow.trade = https://gnosis-protocol.io
```
 - can be checked in `Networks` tab as well

### CODE REVIEW

File: src/custom/utils/explorer.ts [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
+function _getExplorerUrlByEnvironment() {
+  let baseUrl: string | undefined
+  if (isDev) {
+    baseUrl = process.env.REACT_APP_EXPLORER_URL_DEV || 'https://protocol-explorer.dev.gnosisdev.com'
+  } else if (isStaging) {
+    baseUrl = process.env.REACT_APP_EXPLORER_URL_STAGING || 'https://protocol-explorer.staging.gnosisdev.com'
+  } else if (isProd) {
+    baseUrl = process.env.REACT_APP_EXPLORER_URL_PROD || 'https://gnosis-protocol.io'
+  } else {
+    // default to dev
```
Comment (W3stside) [inline src/custom/utils/explorer.ts:16]: I don't believe this should be an error or anything app blocking. Also helpful in localhost

**Comment by W3stside** [comment]
Any idea what this error could be from?:
Screenshot from 2021-04-15 15-08-08

function in question is inside: `custom/utils/environments.ts` - but i don't really see where/why it's being used in tests nor why it's failing. Unless `domainPrefix` is not a `string` when the method is being used

**Comment by W3stside** [comment]
> Any idea what this error could be from?:
> Screenshot from 2021-04-15 15-08-08
> 
> function in question is inside: `custom/utils/environments.ts` - but i don't really see where/why it's being used in tests nor why it's failing. Unless `domainPrefix` is not a `string` when the method is being used

SOLUTION: use `replace` as jest doesn't know about `replaceAll` behaviour is the same for what we need. Concerns?

**Comment by W3stside** [comment]
Questions:
1. do we want PRaul links to be checked and passed with a different environment as `DEV` (default)
2. do we want to check for localhost/0.0.0.0 explicitly?

Otherwise right now the environment checks are only against `cow-trade` / `cow.trade` etc

File: .env [hunk 2]
```diff
@@ -2,4 +2,15 @@ REACT_APP_CHAIN_ID="1"
...
 # Override with your APP_ID
-REACT_APP_ID="0"
\ No newline at end of file
+REACT_APP_ID="0"
+
+# Domain
+# Accept dev as a list of hosts
+REACT_APP_DOMAIN_PREFIX_DEV=cow-trade.dev
+REACT_APP_DOMAIN_PREFIX_STAGING=cow-trade.staging
+REACT_APP_DOMAIN_PREFIX_PROD=cow.trade
```
Comment (alfetopito) [inline .env:11]: Although this is correct right now, it'll change before the release.
Just to keep in mind.
Create an issue maybe? :shrug:

**Review by alfetopito — APPROVED** [review]

File: .env (hunk 2 above)
Comment (W3stside) [reply .env:11]: :+1:

**Review by anxolin — APPROVED** [review]
Cool! I have some suggestion but I went for it and did it in a PR #448

**Comment by anxolin** [comment]
@W3stside I merged here my PR, and there's 2 approvals. So feel free to merge to DEVELOP

**Pull request merged**